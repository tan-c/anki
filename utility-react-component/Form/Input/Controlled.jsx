import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
import withOnKeyDown from '../hoc/withOnKeyDown';

export class InputControlled extends React.Component {
  render() {
    const {
      inputName, inputClassNames, onChange, onKeyDown, inputValue,
    } = this.props;

    return (
      <input
        type="text"
        name={inputName}
        className={inputClassNames}
        ref={(ref) => { this.ref = ref; }}
        value={inputValue}
        onChange={onChange}
        onKeyDown={onKeyDown}
      />
    );
  }
}

InputControlled.defaultProps = {
  inputClassNames: 'flex-1',
  inputName: '',
  inputValue: '',
  onKeyDown: () => {},
  onChange: () => {},
};

InputControlled.propTypes = {
  inputClassNames: PropTypes.string,
  inputName: PropTypes.string,
  inputValue: PropTypes.string,
  onKeyDown: PropTypes.func,
  onChange: PropTypes.func,
};

function mapStateToProps(state, ownProps) {
  return {
    textareaRow: ownProps.textareaRow,
    inputType: ownProps.inputType,
    inputName: ownProps.inputName,
    inputClassNames: ownProps.inputClassNames,
    record: ownProps.record,
  };
}

export default withOnKeyDown(connect(mapStateToProps)(InputControlled), false);
