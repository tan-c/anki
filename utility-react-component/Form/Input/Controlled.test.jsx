import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { InputControlled } from './Controlled';

function setup() {
  const props = {
  };

  return shallow(<InputControlled {...props} />);
}

describe('InputControlled Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find('input')).toHaveLength(1);
    });
  });
});
