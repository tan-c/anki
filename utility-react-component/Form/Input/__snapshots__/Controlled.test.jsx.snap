// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControlled Unit Tests when normally set up renders correctly 1`] = `
<input
  className="flex-1"
  name=""
  onChange={[Function]}
  onKeyDown={[Function]}
  type="text"
  value=""
/>
`;
